Master-Game/Managers/EntityManager.cs

```diff
@@ -24,5 +24,18 @@ public BaseEntity GetPlayer()
             }
             return null;
         }
+
+        public void SetPlayer(PlayerEntity newPlayer)
```

Comment:
what is this for?

Reply:
ahh nvm got it.. could just be CreatePlayer. It's doing more than just SET. 
Also what happens to the previous player in the entitylist? Before going to a new map wouldn't you have to reset some of them managers?

That also brings a thought Each Entity probably needs a ItemManager. Because the player will have items and so will enemies.

Reply:
Each time the map starts it resets the player (GameManager.ProcessGame, Line 57 and 48). When I have the game transition from one map to another (GameManager.ProcessGame, line 82), I have the Map class first build the new map (Map.BuildMapTransition). Before the game creates the map, I clear the current map. When it produces the new map, it will reset everything, including the player. So, I made this SetPlayer method to carry over the current player state (e.g., health points) to the new map. 

What's a better way of doing this?